pkgs/development/interpreters/python/python-packages-base.nix

```diff
@@ -32,20 +32,56 @@ let
       }
       else result);
 
+  preserveFunctionOverride = decorate: f:
+    let
+      fResult = decorate f;
+    in
+    (if (!builtins.isAttrs fResult) then
+      lib.mirrorFunctionArgs f fResult
+    else fResult)
+    // lib.optionalAttrs (f?override) {
+      override = lib.mirrorFunctionArgs f.override
+        (fdrv: decorate (f.override fdrv));
+    };
+
+  compatCustomStdenv = f: origArgs: (
+    if origArgs?stdenv then
+      f.override { inherit (origArgs) stdenv; }
+    else
+      f
+  ) origArgs;
+
   mkPythonDerivation = if python.isPy3k then
     ./mk-python-derivation.nix
   else
     ./python2/mk-python-derivation.nix;
 
-  buildPythonPackage = makeOverridablePythonPackage (lib.makeOverridable (callPackage mkPythonDerivation {
+  # This ensures that the function argument of buildPython* is always mirrored
+  # and that buildPython*.override is always preserved.
+  decorateBuildPyhon = f: lib.pipe f (map preserveFunctionOverride [
```

Comment:
I suppose this is a typo:
decorateBuildPyhon ->  decorateBuildPy**t**hon

Reply:
Ahh! You're right. Thanks!